### PR TITLE
feat: Add a CI for building and pushing docker image to GHCR

### PR DESCRIPTION
**Pull request by asmijafar20**
## Description

Previously the Moja Global docker images were pushed on the docker hub from different people with their accounts. So it's not centralized in one place that's why we have used GitHub Container Registry to maintain all the images in one place. Also, there are no Continuous integration workflows set up in the repository so the images are not able to update when there is a new release in the project. To solve all these problems I have created the CI workflow which will continuously build and publish the image whenever the release triggers it.

- Added the `docker-ci.yml` workflow for building and pushing the docker image on the GitHub container registry using GitHub Actions.
- The workflow runs when the release event triggers.
- Two jobs for `rest-api-flint.example` and `rest-api-gcbm` docker images.

Related: moja-global/FLINT-UI#71

### CODE REVIEW

**Comment by iamrajiv** [comment]
@asmijafar20 I have pulled to code and merged to my repository master branch [here](https://github.com/iamrajiv/FLINT.Cloud)to check whether the packages are publishing or not and I found it's not publishing.

**Comment by iamrajiv** [comment]
@asmijafar20 I have pulled to code and merged to my repository master branch [here](https://github.com/iamrajiv/FLINT.Cloud) to check whether the packages are publishing or not and I found it's not publishing. Can you refer to this https://github.com/moja-global/FLINT.Example/pull/25?

**Comment by asmijafar20** [comment]
Just for checking

**Comment by asmijafar20** [comment]
checking

**Comment by asmijafar20** [comment]
> @asmijafar20 I have pulled to code and merged to my repository master branch [here](https://github.com/iamrajiv/FLINT.Cloud) to check whether the packages are publishing or not and I found it's not publishing. Can you refer to this [moja-global/FLINT.Example#25](https://github.com/moja-global/FLINT.Example/pull/25)?

Review it.

File: .github/workflows/docker-ci.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# This workflow uses actions that are not certified by GitHub.
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:1]: Remove this comment

File: .github/workflows/docker-ci.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+name: Create and publish a Docker image
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:2]: Keep a meaningful name. Suggested: `Publish FLINT.Cloud images`

File: .github/workflows/docker-ci.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+on:
+  push:
+    branches: ["release"]
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:6]: We would like the Docker images to be updated with every push. We don't have a "release" branch yet. Replace it with the "master" branch.

File: .github/workflows/docker-ci.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  build-rest-api-flint-example:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:22]: Add Docker Buildx here.

```sh
       - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
```

File: .github/workflows/docker-ci.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:27]: Suggest change to `${{ github.repository_owner }}`. Refer docs here: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

File: .github/workflows/docker-ci.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
```
Comment (HarshCasper) [inline .github/workflows/docker-ci.yml:58]: Suggest change to ${{ github.repository_owner }}. Refer docs here: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you update the file names as well? 

Change the filename of `ci.yml` to `docker-ci.yml` while changing the filename of `docker-ci.yml` to `docker-publish.yml`.

**Pull request merged**